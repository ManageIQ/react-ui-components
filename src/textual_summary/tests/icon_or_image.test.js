import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import IconOrImage from '../icon_or_image';

describe('Icon or Image', () => {
  it('renders image with title and alt if image is passed in', () => {
    const image = shallow(<IconOrImage image="foo/bar.png" title="foo bar title" />);
    expect(toJson(image)).toMatchSnapshot();
  });

  it('renders icon with title and alt if icon and no image are passed in', () => {
    const image = shallow(<IconOrImage icon="fa fa-foobar" title="foo bar title" />);
    expect(toJson(image)).toMatchSnapshot();
  });

  it('renders nothing if no icon and no image are passed in', () => {
    const image = shallow(<IconOrImage icon="" title="foo bar title" />);
    expect(toJson(image)).toMatchSnapshot();
  });

  it('renders icon with background', () => {
    const image = shallow(<IconOrImage icon="fa fa-foobar" title="foo bar title" background="blue" />)
    expect(toJson(image)).toMatchSnapshot();
  });
});
