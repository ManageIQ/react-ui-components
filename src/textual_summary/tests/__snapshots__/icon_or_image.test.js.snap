// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon or Image renders icon with background 1`] = `
<div
  className="backgrounded-icon"
  style={
    Object {
      "background": "blue",
    }
  }
>
  <i
    className="fa fa-foobar"
    title="foo bar title"
  />
</div>
`;

exports[`Icon or Image renders icon with title and alt if icon and no image are passed in 1`] = `
<i
  className="fa fa-foobar"
  title="foo bar title"
/>
`;

exports[`Icon or Image renders image with title and alt if image is passed in 1`] = `
<img
  alt="foo bar title"
  src="foo/bar.png"
  title="foo bar title"
/>
`;

exports[`Icon or Image renders nothing if no icon and no image are passed in 1`] = `""`;
